lib/WorkerHandler.js

```diff
@@ -80,26 +80,39 @@ function WorkerHandler(script) {
   }
 
   var me = this;
-  this.worker.on('message', function (response) {
-    // find the task from the processing queue, and run the tasks callback
-    var id = response.id;
-    var task = me.processing[id];
-    if (task) {
-      // remove the task from the queue
-      delete me.processing[id];
-
-      // test if we need to terminate
-      if (me.terminating) {
-        // complete worker termination if all tasks are finished
-        me.terminate();
-      }
 
-      // resolve the task's promise
-      if (response.error) {
-        task.resolver.reject(objectToError(response.error));
-      }
-      else {
-        task.resolver.resolve(response.result);
+  // The ready message is only sent if the worker.add method is called (And the default script is not used)
+  if (!script) {
+    this.worker.ready = true;
+  }
+
+  // queue for requests that are received before the worker is ready
+  this.request_queue = [];
```

Comment:
Can you please rename `request_queue` to `requestQueue`? Or maybe `taskQueue` or something? At least lowerCamelCase :).